packages/Table/src/Table.js

```diff
@@ -24,41 +27,57 @@ const Table = React.forwardRef((props, ref) => {
-    "Table.ColumnDefinition",
-  ]);
+  const { "Table.ColumnDefinition": extractedColumnDefinitions, "Table.Headers": extractedHeaders } = extractChildren(
+    children,
+    ["Table.ColumnDefinition", "Table.Headers"]
+  );
 
   let ColumnDefinitions = extractedColumnDefinitions;
 
+  const hasAtLeastOneHeader = React.useMemo(() => {
+    return ColumnDefinitions.some(columnDefinition => {
+      const { header } = columnDefinition.props;
+      return typeof header !== "undefined" && (typeof header === "function" || typeof header === "string");
+    });
+  }, [ColumnDefinitions]);
+
   if (!Array.isArray(extractedColumnDefinitions)) {
     // when there is only one component extracted function return the element not an array of elements
     ColumnDefinitions = [extractedColumnDefinitions];
   }
 
   return (
     <sc.Table aria-label={a11yText} id={tableId} {...moreProps} ref={ref}>
-      <sc.Thead>
-        <tr>
-          {ColumnDefinitions.map((columnDefinition, columnIndex) => {
-            const { cell, header, sticky, ...moreColumnProps } = columnDefinition.props;
-
-            if (typeof header === "function")
-              return (
-                <sc.TH sticky={sticky} borderType={borderType} key={columnIndex} {...moreColumnProps}>
-                  {header({ header: columnDefinition.props, columnIndex })}
-                </sc.TH>
-              );
-            if (typeof header === "string")
-              return (
-                <sc.TH sticky={sticky} borderType={borderType} key={columnIndex} {...moreColumnProps}>
-                  {header}
-                </sc.TH>
-              );
-
-            throw new Error("Header should be either of type string or function");
-          })}
-        </tr>
-      </sc.Thead>
+      {hasAtLeastOneHeader ? (
+        <sc.Thead>
+          {extractedHeaders}
+          <tr>
+            {ColumnDefinitions.map((columnDefinition, columnIndex) => {
+              const { cell, header, sticky, ...moreColumnProps } = columnDefinition.props;
+
+              if (typeof header === "undefined") return null;
+
+              if (typeof header === "function")
+                return (
+                  <sc.TH sticky={sticky} scope="col" borderType={borderType} key={columnIndex} {...moreColumnProps}>
+                    {header({ header: columnDefinition.props, columnIndex })}
+                  </sc.TH>
+                );
+
+              if (typeof header === "string")
+                return (
+                  <sc.TH sticky={sticky} scope="col" borderType={borderType} key={columnIndex} {...moreColumnProps}>
+                    {header}
+                  </sc.TH>
+                );
+
+              throw new Error("Header should be either of type string or function");
+            })}
+          </tr>
+        </sc.Thead>
+      ) : extractedHeaders ? (
+        <sc.Thead>{extractedHeaders}</sc.Thead>
+      ) : null}
```

Comment:
Conditions like that `? : ? :` is hard to read and understand. 
I'd prefer it that case just move this block into a `renderHeader()` function and use a regular `if else` statements instead

Reply:
Also, there is enough logic (including `hasAtLeastOneHeader`) to consider extracting all that into a separate internal component, I think